pkg/prometheus/statefulset.go

```diff
@@ -518,3 +519,104 @@ func GetStatupProbePeriodSecondsAndFailureThreshold(cfp monitoringv1.CommonProme
 
 	return int32(startupPeriodSeconds), int32(startupFailureThreshold)
 }
+
+func MakeContainerPorts(cpf monitoringv1.CommonPrometheusFields) []v1.ContainerPort {
+	var ports []v1.ContainerPort
+	if !cpf.ListenLocal {
+		ports = []v1.ContainerPort{
+			{
+				Name:          cpf.PortName,
+				ContainerPort: 9090,
+				Protocol:      v1.ProtocolTCP,
+			},
+		}
+	}
+	return ports
+}
+
+func BuildVolumes(
+	p monitoringv1.PrometheusInterface,
+	tlsSecrets *operator.ShardedSecret,
+) ([]v1.Volume, []v1.VolumeMount, []v1.VolumeMount, error) {
+	volumes, promVolumeMounts, err := BuildCommonVolumes(p, tlsSecrets)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	configReloaderVolumeMounts := CreateConfigReloaderVolumeMounts()
+
+	return volumes, promVolumeMounts, configReloaderVolumeMounts, nil
+}
+
+func CreateConfigReloaderVolumeMounts() []v1.VolumeMount {
+	return []v1.VolumeMount{
+		{
+			Name:      "config",
+			MountPath: ConfDir,
+		},
+		{
+			Name:      "config-out",
+			MountPath: ConfOutDir,
+		},
+	}
+}
+
+func BuildWebconfig(
+	cpf monitoringv1.CommonPrometheusFields,
+	p monitoringv1.PrometheusInterface,
+) (monitoringv1.Argument, []v1.Volume, []v1.VolumeMount, error) {
+	var fields monitoringv1.WebConfigFileFields
+	if cpf.Web != nil {
+		fields = cpf.Web.WebConfigFileFields
+	}
+
+	webConfig, err := webconfig.New(WebConfigDir, WebConfigSecretName(p), fields)
+	if err != nil {
+		return monitoringv1.Argument{}, nil, nil, err
+	}
+
+	confArg, configVol, configMount, err := webConfig.GetMountParameters()
+	if err != nil {
+		return monitoringv1.Argument{}, nil, nil, err
+	}
+
+	return confArg, configVol, configMount, nil
```

Comment:
This is equivalent, right?
```suggestion
	return webConfig.GetMountParameters()
```